### PR TITLE
Add support to create Mass from string

### PR DESCRIPTION
**Pull request by drodil**
Again behind the from_str feature. Tests included.

### CODE REVIEW

File: src/mass.rs [hunk 1]
```diff
@@ -252,6 +256,42 @@ impl Measurement for Mass {
...
+            let float_val = caps.get(1).unwrap().as_str();
+            return Ok(
+                match caps.get(2).unwrap().as_str().to_lowercase().as_str() {
+                    "ug" | "μg" => Mass::from_micrograms(float_val.parse::<f64>()?),
+                    "mg" => Mass::from_milligrams(float_val.parse::<f64>()?),
+                    "ct" => Mass::from_carats(float_val.parse::<f64>()?),
+                    "g" => Mass::from_grams(float_val.parse::<f64>()?),
+                    "kg" => Mass::from_kilograms(float_val.parse::<f64>()?),
+                    "t" => Mass::from_metric_tons(float_val.parse::<f64>()?),
+                    "gr" => Mass::from_grains(float_val.parse::<f64>()?),
```
Comment (eldruin) [inline src/mass.rs:281]: Hmm, I am a bit unsure about this one. People may use gr. for grams too.

**Review by eldruin — COMMENTED** [review]
Other than the point about `gr.` (which includes a dot, sure) it looks good to me.
Is there a wikipedia entry or so about all these variants? It could be added to the docs.

File: src/mass.rs (hunk 1 above)
Comment (drodil) [reply src/mass.rs:281]: You are correct. Wikipedia says "The SI does not support the use of abbreviations such as "gr" (which is the symbol for grains)"

Comment (drodil) [reply src/mass.rs:281]: Oh. That's for grains. I think it's correct.

Comment (drodil) [reply src/mass.rs:281]: More info https://en.wikipedia.org/wiki/Gram

Comment (eldruin) [reply src/mass.rs:281]: Ok!

**Pull request merged**